guides/source/active_storage_overview.md

```diff
@@ -129,9 +129,12 @@ amazon:
   secret_access_key: ""
   region: ""
   bucket: ""
+  http_read_timeout: "" # default: 60
+  retry_limit: "" # default: 3
```

Comment:
These are defaults of the client library, aren't they?  In other words, can these defaults change without notice?

Reply:
yes, they are the client library defaults. would it be preferable for rails to set default configuration?

Reply:
@jonathanhefner would love to get my first Rails PR merged, what can I do to make this mergeable?

Reply:
My feeling is we should probably avoid listing defaults that are out of our control.  If they change upstream, people may be confused when reading this documentation.

Also, do you know if `aws-sdk-ruby` will accept strings for `http_read_timeout` and `retry_limit`?  If not, we should probably list the placeholders as integers (i.e. `0` instead of `""`).

Reply:
The `aws-sdk-ruby` documentation specifies that these should be integers. Do you think it's too confusing to provide sample values (`30`) instead of a placeholder sample (`0`)?

I was thinking this over some more and timeout configuration is really variable depending on how the application is using this storage type. For instance, applications can download blobs directly and for larger blobs they could need a longer read timeout. However for applications that stream blobs, a smaller read timeout can be more effective.

Reply:
> For instance, applications can download blobs with a single network request and for larger blobs they could need a longer read timeout. However applications can also stream blobs, so a smaller read timeout can be more effective.

I'm not sure I understand.  `http_read_timeout` is the maximum time to wait for an HTTP response from the S3 service, no?  In other words, it applies when the service is being unresponsive, rather than when the transfer is taking too long due to file size or network speed.

Reply:
yes, you are correct, I mixed up the understanding of that in my head. I've updated the placeholder values and `TIP` message as discussed below.